src/test/java/org/cactoos/io/LoggingOutputTest.java

```diff
@@ -0,0 +1,171 @@
+
+/**
+ * Test case for {@link LoggingOutput}.
+ *
+ * @author Fabricio Cabral (fabriciofx@gmail.com)
+ * @version $Id$
+ * @since 0.29
+ * @checkstyle JavadocMethodCheck (500 lines)
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+@SuppressWarnings(
+    {
+        "PMD.MoreThanOneLogger",
+        "PMD.AvoidDuplicateLiterals"
+    }
+)
+public final class LoggingOutputTest {
+
+    @Test
+    public void logWriteZero() throws IOException {
+        final Logger logger = new FakeLogger();
+        new LengthOf(
+            new TeeInput(
+                new InputOf(""),
+                new LoggingOutput(
+                    () -> new ByteArrayOutputStream(),
+                    "memory",
+                    logger
+                )
+            )
+        ).intValue();
+        MatcherAssert.assertThat(
+            "Can't log zero byte written to memory",
+            logger.toString(),
+            Matchers.containsString("Written 0 byte(s) to memory in ")
+        );
+    }
+
+    @Test
+    public void logWriteOneByte() throws IOException {
+        final Logger logger = new FakeLogger();
+        try (
+            final OutputStream out = new LoggingOutput(
+                () -> new ByteArrayOutputStream(),
+                "memory",
+                logger
+            ).stream()
+        ) {
+            out.write(new BytesOf("a").asBytes());
+        }
+        MatcherAssert.assertThat(
+            "Can't log one byte written to memory",
+            logger.toString(),
+            Matchers.containsString("Written 1 byte(s) to memory in")
+        );
+    }
+
+    @Test
+    public void logWriteText() throws IOException {
+        final Logger logger = new FakeLogger();
+        try (
+            final OutputStream out = new LoggingOutput(
+                () -> new ByteArrayOutputStream(),
+                "memory",
+                logger
+            ).stream()
+        ) {
+            out.write(new BytesOf("Hello, товарищ!").asBytes());
+        }
+        MatcherAssert.assertThat(
+            "Can't log 22 bytes written to memory",
+            logger.toString(),
+            Matchers.containsString("Written 22 byte(s) to memory in")
+        );
+    }
+
+    @Test
+    public void logWriteToLargeTextFile() throws Exception {
+        final Logger logger = new FakeLogger();
+        final Path temp = Files.createTempDirectory("ccts-1");
+        final Path path = temp.resolve("x/y/z/file.txt");
+        new LengthOf(
+            new TeeInput(
+                new ResourceOf("org/cactoos/large-text.txt"),
+                new LoggingOutput(
+                    new OutputTo(path),
+                    "text file",
+                    logger
+                )
+            )
+        ).intValue();
+        MatcherAssert.assertThat(
+            "Can't log write and close operations to text file",
+            logger.toString(),
+            Matchers.allOf(
+                Matchers.not(
```

Comment:
@fabriciofx same here, why both contains() and notcontains()?

Reply:
@amihaiemil The same explanation above.